hstream/src/HStream/Server/Handler.hs

```diff
@@ -361,165 +361,194 @@ sendToClient zkHandle qid sc@SourceConnector {..} ss@streamSend = do
 --------------------------------------------------------------------------------
 -- Subscribe
 
-checkSubscriptions
-  :: Int64    -- ^ timer timeout, ms
-  -> ServerContext
-  -> IO ()
-checkSubscriptions timeout ServerContext{..} = do
-  currentTime <- getCurrentTimestamp
-  Log.debug $ "Timeout fire, checkSubscriptions begin, currentTimeStamp = " <> Log.buildInt currentTime
-  (heapRemain, subReader) <- atomically $ do
-    sHeap <- readTVar subscribeHeap
-    let (outDated, remained) = Map.partition (\time -> currentTime - time >= timeout) sHeap
-    mapM_ (updateReaderStatus subscribedReaders Released) $ Map.keys outDated
-    writeTVar subscribeHeap remained
-    readers <- readTVar subscribedReaders
-    return (remained, readers)
-  Log.debug $ "After checkSubscriptions:\n==heapRemain: " <> Log.buildString (show heapRemain) <> "\n==subReader: " <> Log.buildString (show subReader)
-
 createSubscriptionHandler
   :: ServerContext
   -> ServerRequest 'Normal Subscription Subscription
   -> IO (ServerResponse 'Normal Subscription)
-createSubscriptionHandler ServerContext{..} (ServerNormalRequest _metadata subscription@Subscription{..}) = defaultExceptionHandle' doClean $ do
+createSubscriptionHandler ServerContext{..} (ServerNormalRequest _metadata subscription@Subscription{..}) = do
```

Comment:
still need `defaultExceptionHandler` because `S.getUnderlyingLogId` will throw an exception if the stream does not exist.